src/sql/workbench/parts/objectExplorer/browser/treeSelectionHandler.ts

```diff
@@ -57,24 +63,32 @@ export class TreeSelectionHandler {
 		let specificSelection = selection[0];
 
 		if (this.isMouseEvent(event)) {
-			if (this._lastClicked === specificSelection) {
+			if (this._lastClicked !== undefined) {
 				clearTimeout(this._clickTimer);
-				sendSelectionEvent(event, selection, true);
-				return;
+				let lastSpecificClick = this._lastClicked[0];
+
+				if (lastSpecificClick === specificSelection) {
+					sendSelectionEvent(event, selection, true);
+					return;
+				} else {
+					sendSelectionEvent(event, this._lastClicked, false);
+				}
 			}
-			this._lastClicked = specificSelection;
-		}
+			this._lastClicked = selection;
 
-		this._clickTimer = setTimeout(() => {
-			sendSelectionEvent(event, selection, false);
-		}, 300);
+			this._clickTimer = setTimeout(() => {
+				sendSelectionEvent(event, selection, false);
```

Comment:
My point initially why you would want to set selection to undefined is if
Click A
(lasted clicked will be A)
Wait
(this timeout will trigger)
Click A Again
(lastclicked will equal current selection and will trigger a double click even though it should be a single click)

Reply:
https://github.com/microsoft/azuredatastudio/pull/6761/files/0badd8945abeb0a1fd70640d446f9aa0cb0b702e#diff-a06e62f72e909bc558c988be246bba03R50-R52 This should clear the timeout in the scenario you're mentioning

Reply:
Let me check this again.